### PR TITLE
fix(reputation): fix a bug that relate reputation issuance with witne…

### LINKED ISSUE #1137: Reputation issuance bug
Currently, new reputation issuance is related with the witnesses requirements of the data requests included. Also in the case that they can not ever solved (i.e a extreme high number of witnesses required).

The new reputation issuance should be related with the commits or reveals that produces by the data requests. If not, some attacker could create data requests with high number of witnesses to expire the reputation of the ARS for free

### PR DESCRIPTION
**Pull request by lrubiorod**
…sses instead of revealers

Close #1137

### CODE REVIEW

File: node/src/actors/chain_manager/mod.rs [hunk 1]
```diff
@@ -737,8 +738,8 @@ impl ReputationInfo {
         let dr_pointer = tally_transaction.dr_pointer;
         let dr_state = &data_request_pool.data_request_pool[&dr_pointer];
         let commits = &dr_state.info.commits;
-        let replication_factor = dr_state.data_request.witnesses;
-        self.alpha_diff += Alpha(u32::from(replication_factor));
+        let reveals_count = dr_state.info.reveals.len() as u32;
```
Comment (tmpolaczyk) [inline node/src/actors/chain_manager/mod.rs:741]:
```suggestion
        let reveals_count = u32::try_from(dr_state.info.reveals.len()).unwrap();
```

File: node/src/actors/chain_manager/mod.rs [hunk 2]
```diff
@@ -737,8 +737,8 @@ impl ReputationInfo {
         let dr_pointer = tally_transaction.dr_pointer;
         let dr_state = &data_request_pool.data_request_pool[&dr_pointer];
         let commits = &dr_state.info.commits;
-        let replication_factor = dr_state.data_request.witnesses;
-        self.alpha_diff += Alpha(u32::from(replication_factor));
+        let reveals_count = u32::try_from(dr_state.info.reveals.len()).unwrap();
```
Comment (tmpolaczyk) [inline node/src/actors/chain_manager/mod.rs:740]:
```suggestion
        // 1 reveal = 1 witnessing act
        let reveals_count = u32::try_from(dr_state.info.reveals.len()).unwrap();
```

**Comment by aesedepece** [comment]
Wow that went through the speed track! Any chance of this being risky in any way? Is it tested enough?

I mean, shall we include this into the 0.7.3 release?

**Comment by aesedepece** [comment]
> Wow that went through the speed track! Any chance of this being risky in any way? Is it tested enough?
> 
> I mean, shall we include this into the 0.7.3 release?

@lrubiorod @tmpolaczyk any comments?

**Pull request merged**